lib/_tls_wrap.js

```diff
@@ -453,6 +454,12 @@ TLSSocket.prototype._init = function(socket, wrap) {
   if (process.features.tls_npn && options.NPNProtocols)
     ssl.setNPNProtocols(options.NPNProtocols);
 
+  if (process.features.tls_alpn && options.ALPNProtocols) {
+    // keep reference in secureContext not to be GC-ed
+    ssl._secureContext.alpnBuffer = options.ALPNProtocols;
```

Comment:
In the case that `options.ALPNProtocols` is Array, it is converted to buffer of TLS vector format with `tls.convertALPN`. I made a fix in case that it's type is Buffer.